src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useCallback, useContext } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../context/TodosContext';
+import { TodosFilter } from '../TodosFilter';
+
+export const Footer: React.FC = () => {
+  const { todos, onClearCompleted } = useContext(TodosContext);
+
+  const countUnfinished = useCallback(
+    (listTodo: Todo[]) => {
+      return listTodo.map((todo) => todo.completed).filter((todo) => !todo)
+        .length;
+    },
+    [todos],
+  );
+
+  const countFinished = useCallback(
+    (listTodo: Todo[]) => {
+      return listTodo.map((todo) => todo.completed).filter((todo) => todo)
+        .length;
+    },
+    [todos],
+  );
```

Comment:
You can move these functions to a utils file or something like utils, or just out of the component, and useCallback will be do not need